dali/test/python/test_operator_decoders_image.py

```diff
@@ -90,26 +93,159 @@ def log(img_type, size, device, threads):
 
+# ToDo - check padding behavior
+@pipeline_def
+def create_decoder_slice_pipeline(data_path, device):
+    jpegs, _ = fn.readers.file(file_root = data_path,
+                               shard_id = 0,
+                               num_shards = 1,
+                               name = "Reader")
+
+    # ToDo make window random?
+    images_sliced_1 = fn.decoders.image_slice(jpegs,
+                                              [0.1, 0.1],
+                                              [0.6, 0.6],
+                                              device = device,
+                                              hw_decoder_load = 0.7,
+                                              output_type = types.RGB,
+                                              normalized_anchor=True,
+                                              normalized_shape=True,
+                                              axis_names = None,
+                                              axes = None)
+
+    images = fn.decoders.image(jpegs,
+                               device = device,
+                               hw_decoder_load = 0.7,
+                               output_type = types.RGB)
+    # ToDo make window random?
+    images_sliced_2 = fn.slice(images,
+                               [0.1, 0.1],
+                               [0.6, 0.6],
+                               normalized_anchor=True,
+                               normalized_shape=True,
+                               axis_names = None,
+                               axes = None)
+
+    return images_sliced_1, images_sliced_2
 
-class DecoderPipelineFastIDC(Pipeline):
-    def __init__(self, data_path, batch_size, num_threads, use_fast_idct=False):
-        super(DecoderPipelineFastIDC, self).__init__(batch_size, num_threads, 0, prefetch_queue_depth=1)
-        self.input = ops.readers.File(file_root = data_path,
-                                      shard_id = 0,
-                                      num_shards = 1)
-        self.decode = ops.decoders.Image(device = 'cpu', output_type = types.RGB, use_fast_idct=use_fast_idct)
+def run_decode_slice(path, img_type, batch, device, threads):
+    data_path = os.path.join(test_data_root, path, img_type)
+    pipe = create_decoder_slice_pipeline(data_path=data_path, batch_size=batch, num_threads=threads, device_id=0, device=device, prefetch_queue_depth=1)
+    pipe.build()
+    iters = math.ceil(pipe.epoch_size("Reader") / batch)
+    for _ in range(iters):
+        out_1, out_2 = pipe.run()
+        for img_1, img_2 in zip(out_1, out_2):
+            img_1 = to_array(img_1)
+            img_2 = to_array(img_2)
+            assert np.allclose(img_1, img_2)
 
-    def define_graph(self):
-        inputs, labels = self.input(name="Reader")
-        output = self.decode(inputs)
-        return (output, labels)
+def test_image_decoder_slice():
+    for device in {'cpu', 'mixed'}:
+        for threads in {4}:
+            for size in {10}:
```

Comment:
```suggestion
        threads = 4
        size = 10
```
I'd use plain variable assignment unless we plan to add more values